modules/fileout_netcdf/FONcArrayStructureField.cc

```diff
@@ -67,25 +68,113 @@ FONcArrayStructureField::FONcArrayStructureField( BaseType *b, Array* a, bool is
         || supported_atomic_data_type == dods_url_c
         || supported_atomic_data_type == dods_enum_c || supported_atomic_data_type ==dods_opaque_c) {
         string s = "File out netcdf, only support one-layer of simple int/float fields inside an array of structure.";
         throw BESInternalError(s, __FILE__, __LINE__);
     }
 
     d_a = a;
-    d_varname = b->name();
-    if (b_data_type == libdap::dods_array_c) {
-        d_array_type = FONcUtils::get_nc_type(b->var(), is_netCDF4_enhanced);
-        d_array_type_size = (b->var()->width_ll())/(b->var()->length_ll());
-    }
+    if (supported_atomic_data_type == dods_str_c)
+         handle_structure_string_field(b);
     else {
-        d_array_type = FONcUtils::get_nc_type(b, is_netCDF4_enhanced);
-        d_array_type_size = (b->width_ll())/(b->length_ll());
+         d_varname = b->name();
+         if (b_data_type == libdap::dods_array_c) {
+             d_array_type = FONcUtils::get_nc_type(b->var(), is_netCDF4_enhanced);
+             auto t_b = dynamic_cast<Array *>(b);
+             d_array_type_size = (t_b->width_ll()) / (t_b->length_ll());
+         } else {
+             d_array_type = FONcUtils::get_nc_type(b, is_netCDF4_enhanced);
+             d_array_type_size = (b->width_ll()) / (b->length_ll());
+         }
+         // Need to retrieve dimension information here.
+         // In this version, we only try to get the dimension size right.
+         Array::Dim_iter di = d_a->dim_begin();
+         Array::Dim_iter de = d_a->dim_end();
+         for (; di != de; di++) {
+             int64_t size = d_a->dimension_size_ll(di, true);
+             struct_dim_sizes.push_back(size);
+             total_nelements *= size;
+             FONcDim *use_dim = find_sdim(d_a->dimension_name(di), size);
+             struct_dims.push_back(use_dim);
+         }
+         if (b_data_type == libdap::dods_array_c) {
+             auto db_a = dynamic_cast<Array *>(b);
+             if (!db_a) {
+                 string s = "File out netcdf, FONcArrayStructField was passed a variable that is not a DAP Array";
+                 throw BESInternalError(s, __FILE__, __LINE__);
+             }
+             Array::Dim_iter b_di = db_a->dim_begin();
+             Array::Dim_iter b_de = db_a->dim_end();
+             for (; b_di != b_de; b_di++) {
+                 int64_t size = d_a->dimension_size_ll(b_di, true);
+                 struct_dim_sizes.push_back(size);
+                 total_nelements *= size;
+                 field_nelements *= size;
+                 FONcDim *use_dim = find_sdim(db_a->dimension_name(b_di), size);
+                 struct_dims.push_back(use_dim);
+             }
+         }
+     }
+
+}
+
+size_t
+FONcArrayStructureField::obtain_maximum_string_length( ){
+
+    size_t asf_max_str_len = 0;
+
+    // Obtain the compound_buf; this is for gathering the data for individual fields.
+    vector<BaseType*> compound_buf = d_a->get_compound_buf();
```

Comment:
This could be replaced with 'for (auto &cb : compound_buf) { ...'. Although it's just a suggestion.

Reply:
This can. It is too trivial to trigger another travis test. Will do when I clean up the fileout netCDF module.